_component_design/text-inputs.md

```diff
@@ -13,185 +13,291 @@ last-modified: 2017-08-17
+ * The field remains in a focused state as long as the user inputs text and/or doesnt make a selection outside of the field
+ * As the user types into the text field, input text replaces any placeholder text
+
+{% endcolumn %}
+
+{% column right:"hxCol-8 hxCol-xs-12 hxCol-sm-12 hxCol-md-8 hxCol-lg-8" %}
+
+{% figure [class:"image bg-light border"] %}
+![]({{site.url}}/assets/images/components/inputs-and-controls/text-fields/text-fields-text-input-focused.svg)
+{% endfigure %}
+
+{% endcolumn %}
 </div>
 
-## Error
-
-<div class="ui form error">
-  <div class="field error">
-    <label>Error with value</label>
-    <input type="text" value="Value text">
-    <div class="ui error message">
-      <p>There is an error with the response to this question.</p>
-    </div>
-  </div>
-  <div class="field required error">
-    <label>Error with placeholder text</label>
-    <input type="text" placeholder="Required">
-    <div class="ui error message">
-      <p>There is an error with the response to this question.</p>
-    </div>
-    <div class="ui error message icon">
-      <i class="warning circle icon"></i>
-    </div>
-  </div>
-  <div class="field error">
-    <label>Error with icon</label>
-    <input type="text" value="Value text">
-    <div class="ui error message">
-      <p>There is an error with the response to this question.</p>
-    </div>
-  </div>
+<div class="hxRow">
+{% column left:"hxCol-4 hxCol-xs-12 hxCol-sm-12 hxCol-md-4 hxCol-lg-4" %}
+
+### Invalid
+ * The input border changes color and thickness to reflect the invalid state
+ * Error text should be stacked below any helper text
+ * The error message must be preceded by the error icon
+ * Error text should fit on a single line
+ * When focused, the input box gains a shadow, matching the red error color
+ * Once the input contains valid data, it returns to default state and the error message is removed
+
+{% endcolumn %}
+
+{% column right:"hxCol-8 hxCol-xs-12 hxCol-sm-12 hxCol-md-8 hxCol-lg-8" %}
+
+{% figure [class:"image bg-light border"] %}
+![]({{site.url}}/assets/images/components/inputs-and-controls/text-fields/text-fields-text-input-invalid.svg)
+{% endfigure %}
+
+{% endcolumn %}
 </div>
 
-# Textarea
-
-## Enabled
-
-<div class="ui form">
-  <div class="field">
-    <label>Textarea with placeholder text</label>
-    <textarea placeholder="This is placeholder text"></textarea>
-  </div>
-  <div class="field">
-    <label>The final countdown</label>
-    <textarea maxlength="50" id="textarea"></textarea>
-    <div class="ui message subtext right">
-      <p id="textarea_feedback">50 characters remaining</p>
-    </div>
-  </div>
+## Input types
+
+<div class="hxRow">
+{% column left:"hxCol-8 hxCol-xs-12 hxCol-sm-12 hxCol-md-8 hxCol-lg-8" %}
+
+Text fields can be formatted to indicate the types of data a field accepts, using placeholder text, preset layout, and character limits.  Text formatting should be displayed only when the field is focused or filled.
```

Comment:
Text fields can be formatted to indicate the types of data a field accepts by using placeholder text, a preset layout, and character limits.

Reply:
👍